modules/core/test/test_misc.cpp

```diff
@@ -13,6 +13,10 @@
 #include <thread>
 #endif
 
+#ifdef _MSC_VER
+# pragma warning(disable:4702)  // unreachable code
```

Comment:
Do we still need this with `FAIL()`?

Reply:
I wanted to keep it to verify that test body won't start in case of skip exception in the constructor.

Reply:
Do we still need to disable this warning?

Reply:
Only on windows there is a warning: https://pullrequest.opencv.org/buildbot/builders/precommit_windows64/builds/102522. And it points to line in the constructor, not test body: https://github.com/opencv/opencv/pull/24186/files#diff-4a53133499f4421656d6e02cc691c986d7ac26c94235f7480d81c156aebdab07R928

Reply:
It is dangerous to disable this warning for whole file.
Please try to disable it locally through warning push/pop.

Reply:
`CV_NORETURN` solved this issue

Reply:
There is warning on MSVS 2019: https://pullrequest.opencv.org/buildbot/builders/precommit_opencl/builds/100186

Reply:
Added `if` condition to avoid warnings.